libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -702,13 +660,17 @@ class kv_table {
          return return_values;
       }
 
+      virtual bool is_unique() = 0;
```

Comment:
Yes, one of the intents behind this PR was to support secondary indices that are unique.

Reply:
How do you handle `put` for an item that has a secondary key that conflicts with another existing item?

Reply:
Currently, it will update that index to point to the primary key of the new item that was `put`. That is effectively the same behaviour as what `put` does for the primary key.

I can see an argument for checking and throwing an error for secondary keys though, since that behaviour is probably non-obvious.

Reply:
The secondary indices need to be kept consistent with the primary index.  Given only the primary table it should be possible to derive the complete state of all secondary indices.

Reply:
The latest commit should address this and the non-working erase mentioned below.